src/chiltepin/jedi/qg/install.py

```diff
@@ -0,0 +1,101 @@
+
+from parsl.app.app import bash_app, join_app
+
+
+@bash_app(executors=["service"])
+def _clone(env, install_path, tag="develop", stdout=None, stderr=None):
+    return env + textwrap.dedent(
+        f"""
+    echo Started at $(date)
+    echo Executing on $(hostname)
+    git lfs install --skip-repo
+    rm -rf {install_path}/jedi-bundle/{tag}/src
+    mkdir -p {install_path}/jedi-bundle/{tag}
+    cd {install_path}/jedi-bundle/{tag}
+    git clone --branch {tag} https://github.com/JCSDA/jedi-bundle.git src
+    echo Completed at $(date)
+    """
+    )
+
+
+@bash_app(executors=["service"])
+def _configure(env, install_path, tag="develop", stdout=None, stderr=None, clone=None):
+    return env + textwrap.dedent(
+        f"""
+    echo Started at $(date)
+    echo Executing on $(hostname)
+    cd {install_path}/jedi-bundle/{tag}
+    rm -rf build
+    mkdir build
+    cd build
+    # Comment out the parts of the bundle we do not need
+    perl -p -i -e 's/(.* PROJECT vader)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT saber)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT ioda)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT crtm)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT ufo)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT gsw)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT fv3)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT femps)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT soca)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT mom6)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.* PROJECT coupling)/#\\1/g' ../src/CMakeLists.txt
+    perl -p -i -e 's/(.*mpas)/#\\1/ig' ../src/CMakeLists.txt
+    ecbuild -DCMAKE_INSTALL_PREFIX=../ ../src
+    echo Completed at $(date)
+    """
+    )
+
+
+@bash_app(executors=["serial"])
+def _make(
+    env,
+    install_path,
+    tag="develop",
+    jobs=8,
+    stdout=None,
+    stderr=None,
+    configure=None,
+    parsl_resource_specification={"num_nodes": 1},
+):
+    return env + textwrap.dedent(
+        f"""
+    echo Started at $(date)
+    echo Executing on $(hostname)
+    spack env list
+    spack env deactivate
+    cd {install_path}/jedi-bundle/{tag}/build
+    make -j{jobs} VERBOSE=1
+    make install
+    echo Completed at $(date)    """
+    )
+
+
+@join_app
+def run(environment, install_path, tag="develop", jobs=8, stdout=None, stderr=None):
+
+    clone = _clone(
+        environment, install_path=install_path, tag=tag, stdout=stdout, stderr=stderr
+    )
+
+    configure = _configure(
+        environment,
+        install_path=install_path,
+        tag=tag,
+        stdout=stdout,
+        stderr=stderr,
+        clone=clone,
+    )
+
+    make = _make(
+        environment,
+        install_path=install_path,
+        tag=tag,
+        jobs=jobs,
+        stdout=stdout,
+        stderr=stderr,
+        configure=configure,
```

Comment:
Interesting that we have to pass almost the same arguments every time to each step of the process. Hopefully in the future (if possible) we can try to DRY this out

Reply:
We can talk about this.  There are some restrictions when using Parsl Apps that prevent you from doing things that would help dry out the code.  For example, you cannot reference `self` in a Parsl App.  So, while a normal thing to do would be to set those values as a class instance variable, that won't work here because the functions can access `self`.